libraries/tock-cells/src/map_cell.rs

```diff
@@ -76,22 +63,22 @@ impl<T> MapCell<T> {
             return None;
         } else {
             self.occupied.set(false);
-            unsafe { Some(ptr::replace(self.val.get(), mem::uninitialized()).some) }
+            unsafe { Some(ptr::replace(self.val.get(), MaybeUninit::<T>::uninit()).assume_init()) }
```

Comment:
This appear problematic (since `assume_init`) is undefined when called on an uninitialized `MaybeUninit`) but is actually fine because:

This is actually:

```rust
Some(
    ptr::replace(
        self.val.get(), MaybeUninit::<T>::uninit()
    ).assume_init()
)
```

So `assume_init()` is called on the value of type `MaybeUninit<T>` that _used_ to be stored in the pointer but has now been copied out to a temporary (unnamed) variable. And we're verifying it is `occupied` in the if-statement (which is our way of tracking that the value was initialized). We could have rewritten this as:

```rust
let result: MaybeUninit<T> = ptr::replace(self.val.get(), MaybeUninit::<T>::uninit())
// `result` is _initialized_ and now `self.val` is now a new uninitialized value
Some(result.assume_init())
```